eng/common/scripts/update-docs-metadata.ps1

```diff
@@ -70,12 +70,15 @@ function GetAdjustedReadmeContent($pkgInfo, $lang){
     }
 
     $fileContent = $pkgInfo.ReadmeContent
-    $foundTitle = ""
 
     # only replace the version if the formatted header can be found
     $titleRegex = "(\# Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C))"
-    $fileContent = $pkgInfo.ReadmeContent -replace $titleRegex, "`${1} - Version $($pkgInfo.PackageVersion) `n"
-    
+    $titleRegex = "(\# Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C))"
+    $fileTitle = ""
+    if ($pkgInfo.ReadmeContent -match $titleRegex) {
+      $fileContent = $pkgInfo.ReadmeContent -replace $titleRegex, "`${1} - Version $($pkgInfo.PackageVersion) `n"
+      $fileTitle = $Matches[0].Substring(2)
```

Comment:
Is it guaranteed that `Matches` will have at least 1 element that is at least as long as 2 characters? If not, should there be some checks?

If yes, add a comment.

Reply:
I will add comments here. To answer your question, the if check guarantees the Matches always have at least 1 element which start with "# Azure.."

Reply:
Lets not make assumptions on exact number of chars, instead lets use a named group to get what we want. 

So lets change:
`(\# Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C))`
to:
`(\#.*(<filetitle>?Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C)))`
Then you can accesss:
`$fileTitle = $matches["filetitle"]`

Reply:
Fixed it and tested it local. 
Working like 
```
 "(\#.*(?<filetitle>Azure .+? (?:client|plugin|shared) library for (?:JavaScript|Java|Python|\.NET|C)))"
```